### PR TITLE
Updated TUISearchBar with SearchText Publisher

### PR DESCRIPTION
**Pull request by maheshwaran01m**
- Created searchText publisher

### CODE REVIEW

**Review by gopinathcholan — COMMENTED** [review]
Please describe why it is required?

File: Sources/TarkaUI/Components/Search Bar/SearchTextField.swift [hunk 1]
```diff
@@ -36,10 +36,12 @@ struct SearchTextField: View {
   private func performSearch() {
     guard searchBarVM.needDelaySearch else { return }
     searchBarVM.onEditing(searchBarVM.searchItem.text)
+    searchBarVM.searchText = searchBarVM.searchItem.text
   }
   
   private func updateSearchText(_ value: String) {
     searchBarVM.onEditing(value)
```
Comment (arvindhsukumar) [inline Sources/TarkaUI/Components/Search Bar/SearchTextField.swift:43]: Do we need `onEditing` now that there's a published value?

Comment (maheshwaran01m) [reply Sources/TarkaUI/Components/Search Bar/SearchTextField.swift:43]: @arvindhsukumar User can pick there needs, also I don't want to break the existing implementation

**Comment by maheshwaran01m** [comment]
@gopinathcholan  we can resolve this issue

``` swift
 let viewModel = ExampleViewModel()
 
 _searchBarVM = StateObject(wrappedValue: .init(
      searchItem: .init(
        placeholder: viewModel.searchPlaceholder,
        text: viewModel.searchText),
      needDelaySearch: viewModel.needsDelaySearch) {
        viewModel.searchText = $0
      })
      
 _viewModel = StateObject(wrappedValue: viewModel)
 ```

File: Sources/TarkaUI/Components/Search Bar/TUISearchBarViewModel.swift [hunk 2]
```diff
@@ -40,4 +42,14 @@ public class TUISearchBarViewModel: ObservableObject {
...
       self.onEditing = onEditing
     }
+  
+  public init(
+    searchItem: TUISearchBarItem,
+    isShown: Bool = false,
+    needDelaySearch: Bool = false) {
+      self.needDelaySearch = needDelaySearch
+      _searchItem = Published(initialValue: searchItem)
+      _isShown = Published(initialValue: isShown)
```
[the remark this answers is not in the document]
Comment (maheshwaran01m) [reply Sources/TarkaUI/Components/Search Bar/TUISearchBarViewModel.swift:52]: @arvindhsukumar Recently I have found some issue with that, it's better we go with `default init`

**Review by gopinathcholan — APPROVED** [review]

**Review by arvindhsukumar — APPROVED** [review]

**Pull request merged**